packages/esm-ward-app/src/hooks/usePatientPendingOrders.ts

```diff
@@ -0,0 +1,95 @@
+  'custom:(uuid,display,name,datatype,set,answers,hiNormal,hiAbsolute,hiCritical,lowNormal,lowAbsolute,lowCritical,units,' +
+  'setMembers:(uuid,display,answers,datatype,hiNormal,hiAbsolute,hiCritical,lowNormal,lowAbsolute,lowCritical,units))';
+
+/**
+ * Custom hook to count orders of type TestOrder that lack associated observations.
+ *
+ * This hook takes an array of orders and counts how many of them do not have observations recorded.
+ * It fetches the necessary concept and encounter data for each order to determine if observations are missing.
+ *
+ * @param {Order[]} orders - An array of order objects. Each order contains information about the order, including concept and encounter UUIDs.
+ * @returns {count: number, isLoading: boolean } - The count of orders without observations, and a loading state.
+ *
+ * @example
+ * const orders = [
+ *   { concept: { uuid: 'concept-uuid-1' }, encounter: { uuid: 'encounter-uuid-1' }, ... },
+ *   { concept: { uuid: 'concept-uuid-2' }, encounter: { uuid: 'encounter-uuid-2' }, ... },
+ * ];
+ *
+ * const {count: unobservedCount, isLoading } = useCountTestOrdersWithoutObs(testOrders);
+ * // unobservedCount would be the number of test orders without associated observations.
+ */
+
+export function useCountTestOrdersWithoutObs(orders: Order[]): { count: number; isLoading: boolean } {
+  const [countPending, setCountPending] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchOrderData = async (order: Order) => {
+      const conceptKey = `${restBaseUrl}/concept/${order.concept.uuid}?v=${conceptRepresentation}`;
+      const encounterKey = `${restBaseUrl}/encounter/${order.encounter.uuid}?v=${encounterRepresentation}`;
+
+      const [concept, encounter] = await Promise.all([
+        openmrsFetch<Concept>(conceptKey),
+        openmrsFetch<Encounter>(encounterKey),
+      ]);
+
+      return { concept, encounter };
+    };
+
+    const countPendingOrders = async () => {
+      setIsLoading(true);
+      try {
+        const pendingCounts = await Promise.all(
+          orders.map(async (order) => {
+            if (order.orderType['javaClassName'] === 'org.openmrs.TestOrder') {
+              const { concept, encounter } = await fetchOrderData(order);
+
+              if (concept && encounter) {
+                const conceptData = concept.data;
+                const encounterData = encounter.data;
+
+                const testResultObs = encounterData?.obs?.find((obs) => obs.concept.uuid === conceptData.uuid);
+
+                if (conceptData.setMembers.length > 0) {
+                  return conceptData.setMembers.filter(
+                    (memberConcept: { uuid: string }) =>
+                      !testResultObs?.groupMembers?.find((obs) => obs.concept.uuid === memberConcept.uuid),
+                  ).length;
+                } else if (conceptData.setMembers.length === 0 && !testResultObs?.value['display']) {
```

Comment:
What's that second condition, `!testResultObs?.value['display']`, about?

Reply:
The  condition checks if the `testResultObs` has no `groupMembers`. If so, it verifies the `testResultObs` value to determine whether the lab order is done or pending.

I borrowed this implementation from [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/59934a3330a02eb0e41e79df4cabedacdda77f52/packages/esm-patient-orders-app/src/components/test-order.component.tsx#L51)  cc @brandones